project/src/main/java/com/google/sps/servlets/RecommendationServlet.java

```diff
@@ -26,35 +28,49 @@
 public class RecommendationServlet extends HttpServlet {
   private final Gson gson = new Gson();
   private static final Logger LOGGER = Logger.getLogger(RecommendationServlet.class.getName());
+  private static Set<Restaurant> restaurantSet;
+  private static Map<Restaurant, Double> restaurantScores;
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Set<Restaurant> restaurantSet = new HashSet<>();
+    restaurantSet = new HashSet<>();
+    restaurantScores = new HashMap<>();
```

Comment:
We can probably just initialize this above `private static Map<Restaurant, Double> restaurantScores = new HashMap<>();`

Reply:
I think we want to initialize it to empty every time doPost is called or else it doesn't reset if the user enters different inputs and doPost gets called again, if that makes sense.

Reply:
Lets make it a local variable then -- I think one issue with declaring it static is that it becomes a shared instance across all RecommendationServlet objects that are instantiated, based on https://javaee.github.io/javaee-spec/javadocs/javax/servlet/http/HttpServlet.html   it looks like there's no reason for multiple instances of a servlet to exist as it seems to be tied to the lifecycle of the server itself , but if this is something that we need to re-create per-post request, its easier to avoid unintended sideeffects by keeping it scoped within doPost